core/src/main/scala/io/finch/EndpointResult.scala

```diff
@@ -86,9 +59,19 @@ object EndpointResult {
 
     /** Returns the [[Output]] if an [[Endpoint]] is matched.
       */
-    final def output: Option[F[Output[A]]] = self match {
-      case EndpointResult.Matched(_, _, out) => Some(out)
-      case _                                 => None
+    final def outputAttempt(implicit F: ApplicativeError[F, Throwable]): F[Either[Throwable, Output[A]]] =
+      F.attempt(output)
+
+    final def output: F[Output[A]] = self match {
+      case EndpointResult.Matched(_, _, out) => out
+      case _                                 =>
+        throw new IllegalStateException("Endpoint didn't match")
```

Comment:
I would rather raise an error than throw it.
```suggestion
      case _                                 => F.raiseError(new IllegalStateException("Endpoint didn't match"))
```